lib/client/renderer.js

```diff
@@ -417,27 +431,354 @@ function init (client, d3) {
+              'font-size': 15,
+              'font-weight': 'bold',
+              fill: 'white',
+              'text-anchor': 'middle',
+              dy: '.35em'
+            })
+            .text(translate('Move carbs'));
+          chart().drag.append('rect')
+            .attr({
+              class:'drag-droparea',
+              x: insulinRect.x,
+              y: insulinRect.y,
+              width: insulinRect.width,
+              height: insulinRect.height,
+              fill: '#0099ff',
+              opacity: 0.4,
+              rx: 10,
+              ry: 10
+            });
+          chart().drag.append('text')
+            .attr({
+              class:'drag-droparea',
+              x: insulinRect.x + insulinRect.width / 2,
+              y: insulinRect.y + insulinRect.height / 2,
+              'font-size': 15,
+              'font-weight': 'bold',
+              fill: '#0099ff',
+              'text-anchor': 'middle',
+              dy: '.35em'
+            })
+            .text(translate('Move insulin'));
+        }
+          
+        chart().basals.attr('display','none');
+          
+        operation = 'Move';
+      })
+      .on('drag', function() { 
+        //console.log(d3.event);
+        client.tooltip.transition().style('opacity', .9);
+        var x = Math.min(Math.max(0, d3.event.x), chart().charts.attr('width'));
+        var y = Math.min(Math.max(0, d3.event.y), chart().focusHeight);
+        
+        operation = 'Move';
+        if (isInRect(x, y, deleteRect) && isInRect(x, y, insulinRect)) {
+          operation = 'Remove insulin';
+        } else if (isInRect(x, y, deleteRect) && isInRect(x, y, carbsRect)) {
+          operation = 'Remove carbs';
+        } else if (isInRect(x, y, deleteRect)) {
+          operation = 'Remove';
+        } else if (isInRect(x, y, insulinRect)) {
+          operation = 'Move insulin';
+        } else if (isInRect(x, y, carbsRect)) {
+          operation = 'Move carbs';
+        }
+
+        newTime = new Date(chart().xScale.invert(x));
+        var minDiff = times.msecs(newTime.getTime() - treatment.mills).mins.toFixed(0);
+        client.tooltip.html(
+          '<b>' + translate('Operation') + ':</b> ' + translate(operation) + '<br>' 
+          + '<b>' + translate('New time') + ':</b> ' + newTime.toLocaleTimeString() + '<br>' 
+          + '<b>' + translate('Difference') + ':</b> ' +  (minDiff > 0 ? '+' : '') + minDiff + ' ' + translate('mins')
+          );
+
+        chart().drag.selectAll('.arrow').remove();
+        chart().drag.append('line')
+          .attr({
+            'class':'arrow',
+            'marker-end':'url(#arrow)',
+            'x1': chart().xScale(new Date(treatment.mills)),
+            'y1': chart().yScale(client.sbx.scaleEntry(treatment)),
+            'x2': x,
+            'y2': y,
+            'stroke-width': 2,
+            'stroke': 'white'
+          });
+          
+      })
+      .on('dragend', function() { 
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (11)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2042010447/issues/source?bid=2452374&fileBranchId=3130808#l631)